cmd/az-log/cmd/node.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/spf13/cobra"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	consts "sigs.k8s.io/azuredisk-csi-driver/pkg/azureconstants"
+)
+
+// nodeCmd represents the node command
+var nodeCmd = &cobra.Command{
+	Use:   "node",
+	Short: "csi-azuredisk2-node",
+	Long:  `csi-azuredisk2-node`,
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(args) == 0 {
+			fmt.Println("node is a required argument for \"node\" command")
+			os.Exit(0)
+		}
+
+		config := getConfig()
+		clientsetK8s := getKubernetesClientset(config)
+		volumes, nodes, requestIds, since, sinceTime, isFollow, isPrevious := GetFlags(cmd)
+		nodeName := args[0]
+
+		pod := GetAzuredisk2Node(clientsetK8s, nodeName)
+		for {
+			currPodName := pod.Name
+			GetLogsByAzDriverPod(clientsetK8s, currPodName, AzureDiskContainer, volumes, nodes, requestIds, since, sinceTime, isFollow, isPrevious)
+			// If in watch mode (--follow) and the pod failover and restarts, keep watching logs from newly created pos in the same node
+			time.Sleep(10 * time.Second)
+
+			pod = GetAzuredisk2Node(clientsetK8s, nodeName)
+			if !isFollow || pod.Name == currPodName {
+				break
+			}
+		}
+	},
+}
+
+func init() {
+	getCmd.AddCommand(nodeCmd)
+}
+
+func GetAzuredisk2Node(clientsetK8s kubernetes.Interface, node string) *v1.Pod {
+	// Get pod name of the azuredisk2-node running on given node
+	pods, err := clientsetK8s.CoreV1().Pods(consts.ReleaseNamespace).List(context.Background(), metav1.ListOptions{
```

Comment:
This release namespace can also be other than "kube-system", so instead of hardcoding it, we should provide it as a parameter in the config file. If none is given, we can default to kube-system.